providers/mlx5/mlx5_vfio.c

```diff
@@ -1297,7 +1297,7 @@ static void init_eq_buf(struct mlx5_eq *eq)
 
 static uint64_t uar2iova(struct mlx5_vfio_context *ctx, uint32_t index)
 {
-	return (uint64_t)((void *)ctx->bar_map + (index * MLX5_ADAPTER_PAGE_SIZE));
+	return (uint64_t)(uintptr_t)((void *)ctx->bar_map + (index * MLX5_ADAPTER_PAGE_SIZE));
```

Comment:
All of these (uint64_t)(uintptr_t) things are redundant, just cas tto uinptr_t and allow the compiler to widen the type as necessary.